cfme/tests/infrastructure/test_webmks_console.py

```diff
@@ -0,0 +1,178 @@
+# -*- coding: utf-8 -*-
+"""Test for WebMKS Remote Consoles of VMware Providers."""
+import pytest
+import imghdr
+import time
+import re
+
+from cfme.infrastructure.provider import InfraProvider
+from cfme.configure.configuration import VMwareConsoleSupport
+from cfme.configure import configuration
+from cfme.common.vm import VM
+from utils import testgen, version, ssh
+from utils.appliance.implementations.ui import navigate_to
+from utils.log import logger
+from utils.conf import credentials
+from utils.providers import ProviderFilter
+from wait_for import wait_for
+
+pytestmark = pytest.mark.usefixtures('setup_provider')
+
+pytest_generate_tests = testgen.generate(
+    gen_func=testgen.providers,
+    filters=[ProviderFilter(classes=[InfraProvider], required_flags=['webmks_console'])],
+    scope='module'
+)
+
+
+@pytest.fixture(scope="function")
+def vm_obj(request, provider, setup_provider, console_template, vm_name):
+    """
+    Create a VM on the provider with the given template, and return the vm_obj.
+
+    Also, it will remove VM from provider using nested function _delete_vm
+    after the test is completed.
+
+    """
+    vm_obj = VM.factory(vm_name, provider, template_name=console_template)
+
+    @request.addfinalizer
+    def _delete_vm():
+        try:
+            vm_obj.delete_from_provider()
+        except Exception:
+            logger.warning("Failed to delete vm `{}`.".format(vm_obj.name))
+
+    vm_obj.create_on_provider(timeout=2400, find_in_cfme=True, allow_skip="default")
+    return vm_obj
+
+
+@pytest.fixture(scope="module")
+def configure_vmware_console_for_test(appliance, provider):
+    """Configure VMware Console to use VNC which is what is required for the HTML5 console."""
+    navigate_to(appliance.server, 'Server')
+
+    settings_pg = VMwareConsoleSupport(
+        appliance=appliance,
+        console_type='VMware WebMKS',
+    )
+    settings_pg.update()
+
+
+@pytest.fixture(scope="session")
+def configure_websocket():
+    """
+    Enable websocket role if it is disabled.
+
+    Currently the fixture cfme/fixtures/base.py,
+    disables the websocket role to avoid intrusive popups.
+    """
+    roles = configuration.get_server_roles()
+    if 'websocket' in roles and not roles['websocket']:
+        logger.info('Enabling the websocket role to allow console connections')
+        roles['websocket'] = True
+        configuration.set_server_roles(**roles)
+        yield
+    roles['websocket'] = False
+    logger.info('Disabling the websocket role to avoid intrusive popups')
+    configuration.set_server_roles(**roles)
```

Comment:
@kedark3 explained to me the reasons behind this in detail offline and I agreed than in current situation (other fixture is setting it to disabled) it's safer to not touch it during teardown.